src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java

```diff
@@ -677,8 +677,8 @@ private Info lookUpRelatedObject(
         String[] steps = localPropertyName.split("\\.");
         // Step back through ancestor property references If starting at a.b.c.d, then look at
         // a.b.c, then a.b, then a
-        for (int i = steps.length - 1; i >= 0; i--) {
-            String backPropName = Strings.join(".", Arrays.copyOfRange(steps, 0, i));
+        for (int len = steps.length; len > 0; len--) {
```

Comment:
Is it correct to remove the " - 1" here?  If localPropertyName is "a.b.c.d", then the backPropName values should be "a.b.c", "a.b" and "a".

Reply:
Hey, as far as I can see, it's doing exactly the same than before:
Instead of
`String backPropName = Strings.join(".", Arrays.copyOfRange(steps, 0, i));`
it does:
`String backPropName = Arrays.stream(steps).limit(len).collect(Collectors.joining("."));`
The key being `Stream.limit()`, which for the left-most step must be `1`, not `0`.

Reply:
Running the old code on "a.b.c.d" would get [ "a.b.c", "a.b", "a", "" ] while your code gets [ "a.b.c.d", "a.b.c", "a.b", "a" ].  The empty string in the old code is useless and shouldn't be there.  Adding back the "- 1" to your changes gets the desired result of [ "a.b.c", "a.b", "a" ].